lib/Event/ReorderEvent.php

```diff
@@ -30,7 +32,7 @@ class ReorderEvent extends AbstractDocumentEvent
      */
     public function __construct($document, $destId, $after)
     {
-        parent::__construct($document);
+        $this->document = $document;
```

Comment:
You can see the use case already in this PR: https://github.com/sulu-io/sulu-document-manager/pull/28/files#diff-759b4f020ccaf54b3361e72dbe3f7dfbR155

This solution is quite simple, and I would not call it over engineered.